finality-aleph/src/party/member.rs

```diff
@@ -44,3 +59,23 @@ pub fn task<
     let handle = spawn_handle.spawn_essential("aleph/consensus_session_member", task);
     Task::new(handle, stop)
 }
+
+fn rotate_saved_unit_files(stash_path: &Path, session_id: u32) -> Result<(File, File), io::Error> {
+    let extension = ".abfst";
+    let session_path = stash_path.join(format!("{}", session_id));
+    fs::create_dir_all(&session_path)?;
+    let index = fs::read_dir(&session_path)
+        .unwrap()
+        .filter_map(|r| r.ok())
+        .filter_map(|x| x.file_name().into_string().ok())
+        .filter_map(|s| u64::from_str(s.strip_suffix(extension)?).ok())
+        .max();
+    let load_path = match index {
+        Some(index) => session_path.join(format!("{}{}", index, extension)),
+        None => "/dev/null".into(),
```

Comment:
why not `io::empty()`?

Reply:
Good point! Since we're already using box dyn anyway, I can just have this function return that instead of `File`, and use `Empty` for the case where it's…uh, empty.

Reply:
Now there is a buffer here instead of file